src/test/java/com/wikia/webdriver/testcases/auth/LoginTests.java

```diff
@@ -1,74 +1,251 @@
+    ArticlePageObject article = openArticleOnDesktop();
+    Helios.deleteAllTokens(User.INVALIDATED_TOKEN_USER_1);
+    article.refreshPageAddingCacheBuster();
+    assertTrue(article.waitForPageReload().getGlobalNavigation().isUserLoggedOut());
   }
 
-  @Test(groups = "Login_userWithoutAValidTokenGetsLoggedOut")
-  @Execute(asUser = User.USER_12)
-  public void userWithoutAValidTokenGetsLoggedOut() {
-    new ArticleContent().clear();
+  @Test(groups = MOBILE)
+  @Execute(asUser = User.INVALIDATED_TOKEN_USER_2)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void userWithoutAValidTokenGetsLoggedOutOnMobile() {
+    ArticlePage article = openArticleOnMobile();
+    Helios.deleteAllTokens(User.INVALIDATED_TOKEN_USER_2);
+    article.refreshPageAddingCacheBuster();
+    assertTrue(article.isUserLoggedOutMobile());
+  }
 
-    ArticlePageObject article = new ArticlePageObject().open();
-    Helios.deleteAllTokens(User.USER_12);
+  @Test(groups = DESKTOP)
+  public void nonexistentUserCannotLogInOnDesktop() {
+    SignInPage signIn = openLoginModalOnDesktop();
+    String nonexistingUsername = String.format("QA_%s", Instant.now().getEpochSecond());
+    signIn.login(nonexistingUsername, user.getPassword());
+    Assertion.assertEquals(signIn.getError(), ERROR_MESSAGE);
+  }
 
-    article.refreshPageAddingCacheBuster();
-    assertTrue(article.getGlobalNavigation().isUserLoggedOut());
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void nonexistentUserCannotLogInOnMobile() {
+    SignInPage signIn = navigateToSignInOnMobile();
+    String nonexistingUsername = String.format("QA_%s", Instant.now().getEpochSecond());
+    signIn.login(nonexistingUsername, user.getPassword());
+    Assertion.assertEquals(signIn.getError(), ERROR_MESSAGE);
+  }
+
+  @Test(groups = DESKTOP)
+  public void userCannotLogInWithInvalidPasswordOnDesktop() {
+    SignInPage signIn = openLoginModalOnDesktop();
+    String invalidPassword = String.format("P@55_%s", Instant.now().getEpochSecond());
+    signIn.login(user.getUserName(), invalidPassword);
+    Assertion.assertEquals(signIn.getError(), ERROR_MESSAGE);
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void userCannotLogInWithInvalidPasswordOnMobile() {
+    SignInPage signIn = navigateToSignInOnMobile();
+    String invalidPassword = String.format("P@55_%s", Instant.now().getEpochSecond());
+    signIn.login(user.getUserName(), invalidPassword);
+    Assertion.assertEquals(signIn.getError(), ERROR_MESSAGE);
+  }
+
+  @Test(groups = DESKTOP)
+  public void userCannotLogInWithBlankUsernameOnDesktop() {
+    SignInPage signIn = openLoginModalOnDesktop();
+    signIn.typeUsername("").typePassword(user.getPassword());
+    assertTrue(signIn.submitButtonNotClickable(), "Submit button should be disabled");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Submit button should be disabled" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)